db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -6886,9 +6889,71 @@ private void initializeCoreInfo() throws StandardException{
+        noncoreInfo[SYSFOREIGNKEYS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSFOREIGNKEYSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSSTATEMENTS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSSTATEMENTSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSFILES_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSFILESRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSALIASES_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSALIASESRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSTRIGGERS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSTRIGGERSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSSCHEMAPERMS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSSCHEMAPERMSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSTABLEPERMS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSTABLEPERMSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSCOLPERMS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSCOLPERMSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSROUTINEPERMS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSROUTINEPERMSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSROLES_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSROLESRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSSEQUENCES_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSSEQUENCESRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSPERMS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSPERMSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSUSERS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSUSERSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSBACKUP_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSBACKUPRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSBACKUPFILESET_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSBACKUPFILESETRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSBACKUPITEMS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSBACKUPITEMSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSBACKUPJOBS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSBACKUPJOBSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSCOLUMNSTATS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSCOLUMNSTATISTICSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSPHYSICALSTATS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSPHYSICALSTATISTICSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSTABLESTATS_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSTABLESTATISTICSRowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSSOURCECODE_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSSOURCECODERowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSDUMMY1_CATALOG_NUM-NUM_CORE]=new TabInfoImpl(new SYSDUMMY1RowFactory(luuidFactory,exFactory,dvf));
+        noncoreInfo[SYSSNAPSHOT_NUM-NUM_CORE]=new TabInfoImpl(new SYSSNAPSHOTSRowFactory(luuidFactory,exFactory,dvf));
```

Comment:
Are we sure, we're not missing any systables? when I run "show tables" on sqlshell, it gives me 33 rows of systables, it looks like there are 29 rows here. Just wanted to double-check

Reply:
@snowspeeder which ones are missing?

Reply:
I just did a diff below, 1st file is the list of tables derived form sqlshell.sh, the other is from this PR...NOTE the table names have also changed

/sqlshell.list /tmp/code.list 
8,10c8
< SYSCOLUMNS
< SYSCOLUMNSTATS
< SYSCONGLOMERATES
---
> SYSCOLUMNSTATISTICS
18c16
< SYSPHYSICALSTATS
---
> SYSPHYSICALSTATISTICS
23d20
< SYSSCHEMAS
29,30c26
< SYSTABLES
< SYSTABLESTATS
---
> SYSTABLESTATISTICS

Reply:
I guess, name change is ok since they are just the classnames here

Reply:
I see the missing in the core system tables, as well they are just not part of nonCore function there...
+1